src/app/containers/ChartbeatAnalytics/canonical/index.jsx

```diff
@@ -20,11 +24,17 @@ const CanonicalChartbeatBeacon = ({
           _sf_async_config.domain = "${domain}";
           _sf_async_config.useCanonical = ${useCanonical};
           _sf_async_config.useCanonicalDomain = ${useCanonical};
-          _sf_async_config.contentType = "${type}";
+          _sf_async_config.title = "${title}";
+          _sf_async_config.type = "${type}";
           _sf_async_config.sections = "${sections}";
-          _sf_async_config.idSync = {
-           bbc_hid: "${cookie}"
-          };
+          if (${hasReferrer}) {
+            _sf_async_config.virtualReferrer = "${referrer}";
+          }
```

Comment:
Won't this mean that L31 will get output into the markup even if there is no referrer?

Reply:
yes @jamesdonoh , it will be in the markup but virtualReferrer won't be added to the config

Reply:
So it will be output but just say `if (false) { _sf_async_config.virtualReferrer = "undefined" }` or similar if there is no referrer? We should not output redundant/unused JS code, it looks untidy and needlessly adds to the number of bytes the user has to download.

Can you think of a different approach to this and the `idSync` property below?

Reply:
thanks @jamesdonoh I've refactored it